### PR TITLE
Add atan2 for DataVector

### PR DESCRIPTION
**Pull request by nilsdeppe**
This should be rebased after #303 is merged, and should only be merged after https://github.com/LLNL/spack/pull/5992 is merged into spack

### CODE REVIEW

File: docs/MainSite/Installation.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ See LICENSE.txt for details.
 * [Charm++](http://charm.cs.illinois.edu/) 6.8 or newer (must be compiled from source)
 * [Git](https://git-scm.com/)
 * BLAS (e.g. [OpenBLAS](http://www.openblas.net))
-* [Blaze](https://bitbucket.org/blaze-lib/blaze/overview)
+* [Blaze](https://bitbucket.org/blaze-lib/blaze/overview) v3.2
```
Comment (wthrowe) [inline docs/MainSite/Installation.md:17]: Is it worth enforcing this in CMake?

Comment (nilsdeppe) [reply docs/MainSite/Installation.md:17]: Yes, I'll have to see if there's a way to do that. I posted a similar question as an issue on the Blaze repo: https://bitbucket.org/blaze-lib/blaze/issues/138/check-blaze-version

**Comment by nilsdeppe** [comment]
This depends on #315 for Blaze's find_package updates. Before this can be merged the buildenv will need to be updated, which I'm working on now.

Spack has merged the PR mentioned in the description.

File: src/Utilities/PointerVector.hpp [hunk 2]
```diff
@@ -6,16 +6,99 @@
 
 #pragma once
 
+#include <cmath>
 #include "Utilities/Blaze.hpp"
```
Comment (kidder) [inline src/Utilities/PointerVector.hpp]: include blank line and comment

**Review by kidder — DISMISSED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@kidder @wthrowe This has now been rebased on develop and should be ready.

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[containers/Dockerfile.buildenv, line 74 at r2](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304#-Ky2YK__pvm2DsyN_Vig:-Ky2YK__pvm2DsyN_Vih:bls75sf) ([raw file](https://github.com/sxs-collaboration/spectre/blob/72f2cc990b8e7b77b04540278a50094a942c66aa/containers/Dockerfile.buildenv#L74)):*
> ```
> # know is stable. This should be updated periodically to update
> # installed packages.
> RUN git checkout c4db3d384ecd7b75ebf92a5ea0f3740c5137a8ab
> ```

is this really the correct hash :stuck_out_tongue: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304)*

**Comment by kidder** [comment]
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/Utilities/PointerVector.hpp, line 13 at r2](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304#-Ky2c3JZaIS-8tDZBM6s:-Ky2c3JZaIS-8tDZBM6t:b4rbc8b) ([raw file](https://github.com/sxs-collaboration/spectre/blob/72f2cc990b8e7b77b04540278a50094a942c66aa/src/Utilities/PointerVector.hpp#L13)):*
> ```cpp
> // The Utilities/Blaze.hpp configures Blaze
> #include "Utilities/Blaze.hpp"
> ```

remove this blank line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304)*

**Comment by nilsdeppe** [comment]
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/Utilities/PointerVector.hpp, line 13 at r2](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304#-Ky2c3JZaIS-8tDZBM6s:-Ky2cNZHMfswFa4fE9ke:bzh34vl) ([raw file](https://github.com/sxs-collaboration/spectre/blob/72f2cc990b8e7b77b04540278a50094a942c66aa/src/Utilities/PointerVector.hpp#L13)):*
<details><summary><i>Previously, kidder (Lawrence Kidder) wrote…</i></summary><blockquote>

remove this blank line
</blockquote></details>

Since this is needed for configuring Blaze, I think it's better to be separate.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304)*

**Comment by kidder** [comment]
Reviewed 5 of 5 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304)*

**Comment by kidder** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sxs-collaboration/spectre/304#-:-Ky2eDUMU9-sq_8QyrFH:bnfp4nl)*

**Review by kidder — CHANGES_REQUESTED** [review]
mac builds are failing because blaze 3.2 is required but only 3.1 is found

**Comment by nilsdeppe** [comment]
Okay, pushed a commit that should fix this

**Review by kidder — APPROVED** [review]

**Pull request merged**